url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/KitRobotAgent.java

```diff
@@ -0,0 +1,461 @@
+				}
+			}
+		}
+
+		// We will always attempt to fill the stand, in case a kit fails
+		// inspection. If the last kit is unneeded, we'll just put it on the
+		// "bad" conveyor.
+		if (kitWaitingOnConveyor && !kitRequested && state != KitRobotState.HOLDING_KIT
+				&& (standPositions.get(1) || standPositions.get(2))) {
+			kitRequested = true;
+			conveyor.msgGiveMeKit();
+			// return true;
+		}
+
+		/*
+		 * Tried all rules and found no actions to fire. Return false to the main loop of abstract base class Agent and
+		 * wait.
+		 */
+		return false;
+	}
+
+	/*
+	 * Actions
+	 */
+
+	/**
+	 * Takes a kit from the conveyor and place it on the stand.
+	 */
+	private void placeKitOnStand(MyKit mk) {
+		print("Attempting to place kit");
+		// Only need to check 1 and 2
+		for (int loc = 1; loc < 3; loc++) {
+			if (standPositions.get(loc) == true) {
+				print("Found location.");
+				standPositions.put(loc, false);
+				mk.location = loc;
+				mk.KS = KitStatus.ON_STAND;
+				if (mockgraphics != null) {
+					mockgraphics.msgPlaceKitOnStand(mk.kit.kitGraphics, loc);
+				}
+				if (kitrobotGraphics != null) {
+					kitrobotGraphics.msgPlaceKitOnStand(mk.kit.kitGraphics, loc);
+				}
+				try {
+					animation.acquire();
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+				// print("Got permit");
+				state = KitRobotState.NOT_HOLDING_KIT;
+				stand.msgHereIsKit(mk.kit, loc);
+				// print("Kit placed. Now asking stand to place kit");
+				break;
+			}
+		}
+		stateChanged();
+	}
+
+	/**
+	 * Places an assembled kit on the stand into the inspection area (also on the stand).
+	 * 
+	 * @param k
+	 *            the kit being placed.
+	 */
+	private void placeKitInInspectionArea(MyKit mk) {
+		standPositions.put(0, false);
+		if (mockgraphics != null) {
+			mockgraphics.msgPlaceKitInInspectionArea(mk.kit.kitGraphics);
+		}
+		if (kitrobotGraphics != null) {
+			if (mk.kit == null) {
+				print("Inspection Kit is null");
+			}
+			if (mk.kit.kitGraphics == null) {
+				print("Inspection KitGraphics null");
+			}
+			kitrobotGraphics.msgPlaceKitInInspectionArea(mk.kit.kitGraphics);
+		}
+		try {
+			animation.acquire();
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		// print("Got permit");
+
+		camera.msgInspectKit(mk.kit);
+
+		stand.msgMovedToInspectionArea(mk.kit, mk.location);
+		stateChanged();
+	}
+
+	/**
+	 * Places a completed kit on the conveyor for removal from the kitting cell.
+	 * 
+	 * @param k
+	 *            the kit being shipped out of the kitting cell.
+	 */
+	private void shipKit(MyKit mk) {
+		print("Removing " + mk.kit.toString());
+		myKits.remove(mk);
+		if (mockgraphics != null) {
+			mockgraphics.msgPlaceKitOnConveyor();
+		}
+		if (kitrobotGraphics != null) {
+			kitrobotGraphics.msgPlaceKitOnConveyor();
+		}
+		try {
+			animation.acquire();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)